Hardcode Istio version in download-istio hack script

mayleighnmyers opened the pull request:
### Describe the change

I removed the 2 hour stalling when there API rate limits, making the hack script fail faster. I also added the istio version to be hardcoded for a fallback if the Github API's do not work/ or not found. 

### Steps to test the PR
Have crc running
run until you get 403's:
```
for i in {1..61}; do
    echo "Request $i..."
    curl -s -o /dev/null -w "%{http_code}" https://api.github.com/repos/istio/istio/releases
    echo
done
```
Rerun `./download-istio.sh `


### Issue reference

Closes https://github.com/kiali/kiali/issues/8530#issuecomment-3424918136

jmazzitelli commented:
Adding a GitHub Auth Token if one is available is nice.

I would just like:

(a) Keep the retries - if for some reason the latest release version cannot be obtained from the GitHub API - retry it however many times it was retrying before. Same for any other retries that were done in this script (not sure if there were more?). This helps the use case where a network outage or a GitHub outage is causing the failure - usually, those go away within 30 minutes or so. We don't want our nightly tests to fail because of that. If you want, you could look at the HTTP response code and if it is a special code that indicates rate limiting is causing the failure, then you can fail fast in that case (that would be a nice to have, not required - I don't know if GitHub API returns a special response code that means "rate limited failure").

(b) Don't blindly rely on a fallback version. If for some reason the latest version cannot be determined, the script should fail with a non-zero exit code. We don't want to run with some version that we aren't expecting or is not appropriate.

As long as those two things are in the script, feel free to add whatever is needed - github auth token or whatever

mayleighnmyers commented:
@jmazzitelli I fixed a local mode cypress test because that is what failed my run, is it okay that I fixed it here, or do you suggest leaving it as is and merging with that error and fixing in a different PR (possibly)?
